src/Kuzzle.ts

```diff
@@ -769,7 +770,13 @@ Discarded request: ${JSON.stringify(request)}`));
    * On token expiration, reset jwt and unsubscribe all rooms.
    * Throttles to avoid duplicate event triggers.
    */
-  tokenExpired() {
+  async tokenExpired () {
+    if (this.authenticator && await this.tryReAuthenticate()) {
```

Comment:
We still have an issue because if the user call `auth.logout`, the `tokenExpired` event will be triggered and thus the authenticator will be used even if the user didn't wanted to re-auth

Reply:
We might need to have a state when we successfully login and set it to false when doing logout